### PR TITLE
fix: add state to controlled components

### PR DESCRIPTION
**Pull request by SwaySway**
*Issue #, if available:*

*Description of changes:*
- adding state to controlled components so they get updated with the default value
- selectField as a default placeholder before the state is updated

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by hein-j — DISMISSED** [review]

**Comment by hein-j** [comment]
Approved, though not sure why you need a placeholder for SelectField

**Review by cwoolum — DISMISSED** [review]
LGTM

File: packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts [hunk 1]
```diff
@@ -214,7 +214,7 @@ export function getFormDefinitionInputElement(
         props: {
           label: config.label || baseConfig?.label || FORM_DEFINITION_DEFAULTS.field.inputType.label,
           descriptiveText: config.inputType?.descriptiveText ?? baseConfig?.inputType?.descriptiveText,
-          placeholder: config.inputType?.placeholder || baseConfig?.inputType?.placeholder,
+          placeholder: config.inputType?.placeholder || baseConfig?.inputType?.placeholder || 'Please select an option',
```
Comment (awinberg-aws) [inline packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts:217]: does this need i18n?

Comment (SwaySway) [reply packages/codegen-ui/lib/generate-form-definition/helpers/form-field.ts:217]: Yea though I think we'll need to address this as a follow up in a larger sense as most text does not support i18n today

**Review by awinberg-aws — DISMISSED** [review]

**Comment by SwaySway** [comment]
> Approved, though not sure why you need a placeholder for SelectField

the placeholder is there to indicate to customers they need selection an option as we don't want to assume which default enum to set for the react state

**Review by cwoolum — APPROVED** [review]
LGTM

**Review by hein-j — APPROVED** [review]

**Comment by hein-j** [comment]
> > Approved, though not sure why you need a placeholder for SelectField
> 
> the placeholder is there to indicate to customers they need selection an option as we don't want to assume which default enum to set for the react state

Got it

**Review by awinberg-aws — APPROVED** [review]

**Pull request merged**